modules/markup/csv/csv.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2018 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package markup
+
+import (
+	"bytes"
+	"encoding/csv"
+	"io"
+
+	"code.gitea.io/gitea/modules/markup"
+)
+
+func init() {
+	markup.RegisterParser(Parser{})
+}
+
+// Parser implements markup.Parser for orgmode
+type Parser struct {
+}
+
+// Name implements markup.Parser
+func (Parser) Name() string {
+	return "csv"
+}
+
+// Extensions implements markup.Parser
+func (Parser) Extensions() []string {
+	return []string{".csv"}
+}
+
+// Render implements markup.Parser
+func (Parser) Render(rawBytes []byte, urlPrefix string, metas map[string]string, isWiki bool) []byte {
+	rd := csv.NewReader(bytes.NewReader(rawBytes))
+	var tmpBlock bytes.Buffer
+	tmpBlock.WriteString(`<table class="table">`)
+	for {
+		fields, err := rd.Read()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			continue
+		}
+		tmpBlock.WriteString("<tr>")
+		for _, field := range fields {
+			tmpBlock.WriteString("<td>")
+			tmpBlock.WriteString(field)
```

Comment:
Shouldn't this be html-escape to prevent XSS?

Reply:
You are right. I will send another commit to fix that.

Reply:
see https://github.com/go-gitea/gitea/blob/master/modules/markup/markup.go#L79, I think all parser will be filtered by that.

Reply:
This one isn't, since you're directly crafting html by hand.

Reply:
Unfortunately, [bluemonday]( https://github.com/microcosm-cc/bluemonday ) is expecting well-formed html, so odds are that the sanitization is trivially bypassable.

Reply:
@lunny any html tags in csv must be rendered as is not interpreted as html